capsules/src/dals/README.md

```diff
@@ -0,0 +1,100 @@
+
+
+The trusted `AppLoader` capsule interfaces with 4 differennt peripheral capsules. These
+peripheral capsules consist of the **AppLoaderClient**, **Decompressor**, **Verifier**, and 
+**Validator** capsules.  Each capsule is named after the trait or interface it should implement.
+These interfaces are defined in `interfaces.rs`.
+
+The way these capsules are interconnected is diagrammed below.
+
+
+                                            AppLoaderClient
+                                                 |
+                                                 |
+                                                 |
+                                                 |
+                         Validator ---------  AppLoader  ----------  Decompressor
+                                                 |
+                                                 |
+                                                 |
+                                                 |
+                                              Verifier
+                                        
+
+There exists a circular reference between the **AppLoader** capsule and each of the peripheral
+capsules.  This enables the **AppLoader** to call functions implemented by a peripheral capsule and 
+vice versa.
+
+The peripheral modules are detailed below.  Please see `interfaces.rs` for more information on functions and such.
+
+
+## AppLoaderClient
+
+This capsule implements `pub trait AppLoaderClient<'a>`.  This capsule can be any capsule
+which receives data via standard communication protocols for embedded systems such as BLE, 
+Zigbee, USB.  This capsule's main function is to send this raw received data buffer-by-buffer
+to the **AppLoader** capsule.  In order to prevent bottleneck issues, this capsule must send an ack 
+only after its buffer is returned to it from the **AppLoader**.
+
+This capsule must also include a buffer the size of one data segment (different for different protocols) which
+it sends to the **AppLoader** to send one segment of the received data.
+
+## Decompressor
+
+**DALS** support data decompression methods which "operate on-the-fly".  That is, the received app data
+is decompressed segment-by-segment. This capsule must implement `pub trait Decompressor<'a>` and contain 
+a decompression algorithm which conforms to the above constraints.
+
+This capsule must include a buffer in which to store the resulting decompressed data which is passed
+back to the **AppLoader**.
+
+
+## Verifier
+
+This capsule implements `pub trait Verifier<'a>`.  This capsule contains an algorithm which can
+perform data verification on the app data loaded into flash.  Such an algorithm could be used to 
+verify that no bit-errors occured during wireless transmission or that the loaded app is not malware.
+Examples of this include checksums, digital signatures or more complex cryptographic computations.
+
+This capsule must contain an immutable reference to the app binary data in flash space.
```

Comment:
```suggestion
This capsule must accept an immutable reference to the app binary data in flash space.
```
Maybe? It seems like this capsule would not always own such a reference but just receive it as part of the interaction with the `AppLoader`